codegen/src/lib.rs

```diff
@@ -19,14 +19,13 @@
 // OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
 // WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
 
-mod definition;
+pub mod definition;
```

Comment:
what if we expose only the generate function?

Reply:
That's precisely why it's public. https://rust-lang.github.io/rust-clippy/master/#module_name_repetitions